tests/integration/utils.py

```diff
@@ -53,3 +54,56 @@ def create_new_request(self, payload):
         )
         resp.raise_for_status()
         return Response(resp.json(), resp.json()["id"], resp.status_code)
+
+    def download_bundle(self, request_id):
+        """
+        Download a bundle archive.
+
+        :param int request_id:  ID of the request in Cachito
+        :return: Object that contains response from the Cachito API
+        :rtype: Response
+        :raises requests.exceptions.HTTPError: if the request to the Cachito API fails
+        """
+        resp = requests.get(f"{self._cachito_api_url}/requests/{request_id}/download")
```

Comment:
These bundles can be quite large and this approach loads the bundle in memory. Could you please consider using the streaming approach as described here?
https://requests.readthedocs.io/en/latest/user/advanced/#body-content-workflow
https://stackoverflow.com/a/16696317/9706416